torch/_inductor/codegen/cpp.py

```diff
@@ -616,8 +649,44 @@ def codegen_define_and_call(self, wrapper):
         wrapper.define_kernel(kernel_name, codecache_str)
 
         # generate the code to call this
+        self.generate_kernel_calling_code(
+            wrapper, kernel_name, call_args, code, arg_types
+        )
+
+
+class CppWrapperKernelGroup(KernelGroup):
+    def __init__(self):
+        super().__init__()
+        self.args = CppWrapperKernelArgs()
+
+    def get_kernel_path(self, code):
+        # TODO: this duplicates with CodeCache logic
+        ext = "so"
+        extra = cpp_compile_command("i", "o")
+        # \n is required to match with the CodeCache behavior
+        source_code = "\n" + code.getvalue()
+        basename = code_hash(source_code + extra)
+        subdir = os.path.join(cache_dir(), basename[1:3])
+        kernel_path = os.path.join(subdir, f"{basename}.{ext}")
+        return kernel_path
+
+    def generate_kernel_calling_code(
+        self, wrapper, kernel_name, call_args, code, arg_types
+    ):
+        kernel_path = self.get_kernel_path(code)
+
         wrapper.writeline(
-            "{}({})".format(kernel_name, ", ".join(call_args)),
+            f'auto {kernel_name}_lib = dlopen("{kernel_path}", RTLD_NOW);'
+        )
+        wrapper.writeline(f"assert({kernel_name}_lib != nullptr);")
+        wrapper.writeline(f"void (*{kernel_name})({arg_types});")
+        wrapper.writeline(
+            f'*(void **) (&{kernel_name}) = dlsym({kernel_name}_lib, "kernel");'
```

Comment:
Should this part (and the dlsym call) be cached?

Reply:
Yes, we will cache this part as the next step.